src/esp/physics/bullet/BulletRigidObject.cpp

```diff
@@ -381,12 +381,34 @@ void BulletRigidObject::constructAndAddRigidBody(MotionType mt) {
 }
 
 void BulletRigidObject::activateCollisionIsland() {
+  btCollisionObject* thisColObj = bObjectRigidBody_.get();
+
+  // first query overlapping pairs of the current object from the most recent
+  // broadphase to collect relevant simulation islands.
+  std::set<int> overlappingSimulationIslands = {thisColObj->getIslandTag()};
```

Comment:
Nit: Really just food for thought: in Cruise's C++ codebase as well as in the game industry, I'm used to seeing an effort to avoid frequent, per-frame dynamic allocations, such as this local set container which must grow dynamically.

Unfortunately I've never seen a really great alternative. Sometimes a codebase might declare this container as static and just clear the size to zero every frame, reusing the memory, but this is error-prone and non multithreading-safe. Sometimes a codebase offers a special heap allocator which is very fast but mostly requires FILO (stack-ordered) allocs and frees.

Reply:
Hopefully this won't be per-frame, but I get what you are saying. 👍